test/e2e/suite-ingress/suite-ingress-features/namespace.go

```diff
@@ -189,3 +269,130 @@ spec:
 		})
 	})
 })
+
+var _ = ginkgo.Describe("suite-ingress-features: namespacing ub-label", func() {
+	labelName, labelValue := fmt.Sprintf("namespace-selector-%d", time.Now().Nanosecond()), "watch"
+	s := scaffold.NewScaffold(&scaffold.Options{
+		Name:                  "un-label",
+		IngressAPISIXReplicas: 1,
+		ApisixResourceVersion: scaffold.ApisixResourceVersion().Default,
+		NamespaceSelectorLabel: map[string]string{
+			labelName: labelValue,
+		},
+		DisableNamespaceLabel: true,
+	})
+	namespace1 := fmt.Sprintf("un-label-%d", time.Now().Nanosecond())
+
+	ginkgo.It("un-label", func() {
+		ns := fmt.Sprintf(`
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+  labels:
+    %s: %s
+`, namespace1, labelName, labelValue)
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromStringWithNamespace(ns, namespace1), "creating namespace")
+		//defer s.DeleteResourceFromStringWithNamespace(ns, namespace1)
+		_, err := s.NewHTTPBINWithNamespace(namespace1)
+		assert.Nil(ginkgo.GinkgoT(), err, "create httpbin service in", namespace1)
+
+		backendSvc, backendSvcPort := s.DefaultHTTPBackend()
+		route1 := fmt.Sprintf(`
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: httpbin-route
+spec:
+  ingressClassName: apisix
+  rules:
+  - host: httpbin.com
+    http:
+      paths:
+      - path: /ip
+        pathType: Exact
+        backend:
+          service:
+            name: %s
+            port:
+              number: %d
+`, backendSvc, backendSvcPort[0])
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromStringWithNamespace(route1, namespace1), "creating ingress")
+		assert.Nil(ginkgo.GinkgoT(), s.EnsureNumApisixRoutesCreated(1))
+		time.Sleep(time.Second * 6)
+		_ = s.NewAPISIXClient().GET("/ip").WithHeader("Host", "httpbin.com").Expect().Status(http.StatusOK).Body().Raw()
+
+		assert.Nil(ginkgo.GinkgoT(), s.DeleteResourceFromStringWithNamespace(route1, namespace1), "deleting ingress")
+		ns = fmt.Sprintf(`
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+`, namespace1)
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ns), "creating namespace")
+		assert.Nil(ginkgo.GinkgoT(), err, "create httpbin service in", namespace1)
+		time.Sleep(6 * time.Second)
+		routes, err := s.ListApisixRoutes()
+		assert.Nil(ginkgo.GinkgoT(), err)
+		assert.Len(ginkgo.GinkgoT(), routes, 0)
```

Comment:
In order to be more in line with the actual situation, I suggest modifying the logic here to directly clean up the namespace labels

Reply:
There is no such interface. This can be done, but it is not convenient.

Reply:
https://pkg.go.dev/k8s.io/client-go@v0.25.0/kubernetes/typed/core/v1#NamespaceInterface

This function can be done through Update or Patch

Reply:
Patch is simpler, just need to pass one of the following

```
 {"metadata":{"labels":{"YOUR Label Name":null}}}
```

Reply:
Done.